### DISCUSSION TITLE
Mention complexity of functions if it is known

### DISCUSSION THREAD

**Discussion by mamekoro** [bevyengine/bevy#10098] [Ideas]
I was looking for information on the performance of `Query` and found the following comment.

> `Query.get` is `O(1)`, and resources are also a constant time lookup :)
> ― https://github.com/bevyengine/bevy/issues/6556#issuecomment-1358516794

What about other functions?

I frequently use query-like functions such as `Query::get()`, `Query::single()`, `Commands::entity()`, `Res<Input<KeyCode>>::just_pressed()`, etc, but I'm just a Bevy user and don't know about the complexity of them.

If the complexity of a function is known, please consider documenting it.

**Comment by alice-i-cecile** [comment]
This is a great idea :) Going to convert it to a discussion for now: feel free to open as many issues as you'd like for specific APIs / clusters of APIs. Otherwise this issue will stay open in a state of limbo forever, as this isn't exactly "completable" in its current forms.

**Reply by mamekoro** [reply]
Thanks, I'll open a new issue.
